### PR TITLE
Check save certificate

### LINKED ISSUE #692: iOS 3.4.9 app crashes: CheckAccessToServer saveCertifcate call to fopen
I am experiencing crashes with the iOS ownCloud app after the upgrade to 3.4.9.

### Steps to reproduce
Unfortunately I haven't been able to reproduce this yet.

### Expected behaviour
App shouldn't crash.

### Actual behaviour
It appears to be caused by the fact that in CheckAccessToServer saveCertifcate a call to fopen is done, but it is never checked whether it is successful. The resulting FILE handle is then handed down into PEM_write_x509.

### Client
**iOS version:**
9.3.2

**ownCloud app version:**
3.4.9

**Device model:**
various devices

I got the following stacktrace from xcode:

```
Thread 0 name:
Thread 0 Crashed:
0 libsystem_c.dylib 0x00000001835737b8 flockfile + 0 (_flock_stub.c:61)
1 libsystem_c.dylib 0x00000001835793c8 fwrite + 76 (fwrite.c:79)
2 Owncloud iOs Client 0x000000010024cd10 file_write + 52
3 Owncloud iOs Client 0x000000010024c644 BIO_write + 124
4 Owncloud iOs Client 0x000000010025a4c4 PEM_write_bio + 108
5 Owncloud iOs Client 0x000000010025a268 PEM_ASN1_write_bio + 468
6 Owncloud iOs Client 0x000000010025a040 PEM_ASN1_write + 144
7 Owncloud iOs Client 0x000000010025a780 PEM_write_X509 + 68
8 Owncloud iOs Client 0x000000010020607c -[CheckAccessToServer saveCertificate:withName:] + 272 (CheckAccessToServer.m:270)
9 Owncloud iOs Client 0x0000000100205d5c -[CheckAccessToServer connection:willSendRequestForAuthenticationChallenge:] + 148 (CheckAccessToServer.m:218)
10 CFNetwork 0x000000018423d6a0 __65-[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:]_block_invoke + 80 (NSURLConnectionInternal.mm:124)
11 CFNetwork 0x000000018423d630 -[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:] + 200 (NSURLConnectionInternal.mm:121)
12 CFNetwork 0x000000018423d7a4 -[NSURLConnectionInternal _withActiveConnectionAndDelegate:] + 56 (NSURLConnectionInternal.mm:139)
13 CFNetwork 0x0000000184210688 -[NSURLConnectionInternalConnection(Internal) sendCFChallenge:toSelector:] + 80 (NSURLConnectionInternalConnection.mm:245)
14 CFNetwork 0x00000001840a9d58 ___ZN27URLConnectionClient_Classic51_delegate_willSendRequestForAuthenticationChallengeEP19_CFURLAuthChallenge_block_invoke + 120 (URLConnectionClient.cpp:2312)
15 CFNetwork 0x000000018419741c ___ZN27URLConnectionClient_Classic18_withDelegateAsyncEPKcU13block_pointerFvP16_CFURLConnectionPK33CFURLConnectionClientCurrent_VMaxE_block_invoke_2 + 108 (URLConnectionClient.cpp:1896)
16 libdispatch.dylib 0x000000018350d47c _dispatch_client_callout + 16 (object.m:506)
17 libdispatch.dylib 0x00000001835167b8 _dispatch_block_invoke + 540 (queue.c:2171)
18 CFNetwork 0x000000018408fca8 RunloopBlockContext::_invoke_block(void const*, void*) + 36 (CoreSchedulingSet.mm:363)
19 CoreFoundation 0x00000001839a067c CFArrayApplyFunction + 68 (CFArray.c:707)
20 CFNetwork 0x000000018408fb8c RunloopBlockContext::perform() + 136 (CoreSchedulingSet.mm:317)
21 CFNetwork 0x000000018408fa4c MultiplexerSource::perform() + 312 (CFNRunLoopMultiplexer.c:282)
22 CFNetwork 0x000000018408f878 MultiplexerSource::_perform(void*) + 68 (CFNRunLoopMultiplexer.c:47)
23 CoreFoundation 0x0000000183a7909c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24 (CFRunLoop.c:1761)
24 CoreFoundation 0x0000000183a78b30 __CFRunLoopDoSources0 + 540 (CFRunLoop.c:1807)
25 CoreFoundation 0x0000000183a76830 __CFRunLoopRun + 724 (CFRunLoop.c:2536)
26 CoreFoundation 0x00000001839a0c50 CFRunLoopRunSpecific + 384 (CFRunLoop.c:2814)
27 GraphicsServices 0x0000000185288088 GSEventRunModal + 180 (GSEvent.c:2245)
28 UIKit 0x0000000188c8a088 UIApplicationMain + 204 (UIApplication.m:3772)
29 Owncloud iOs Client 0x00000001000ddf7c main + 88 (main.m:23)
30 libdyld.dylib 0x000000018353e8b8 start + 4 (start_glue.s:78)
```

@nasli @mRoRo @rperezb Do you need more information to investigate this? /cc @felixboehm 

00005603 (from https://github.com/owncloud/enterprise/issues/1363)

PR: #703

### PR DESCRIPTION
**Pull request by nasli**
Fixes #692

### CODE REVIEW

File: Owncloud iOs Client/DataBase/Queries/ManageAppSettingsDB.m [hunk 1]
```diff
@@ -202,7 +202,7 @@ +(NSMutableArray*) getAllCertificatesLocation {
     
     NSString *documentsDirectory = [UtilsUrls getOwnCloudFilePath];
     
-    NSString *localCertificatesFolder = [NSString stringWithFormat:@"%@/Certificates/",documentsDirectory];
+    NSString *localCertificatesFolder = [NSString stringWithFormat:@"%@Certificates/",documentsDirectory];
```
Comment (mRoRo) [inline Owncloud iOs Client/DataBase/Queries/ManageAppSettingsDB.m]: Probably you could use getLocalCertificatesPath method here

File: Owncloud iOs Client/Network/URLSession Subclasses/OCURLSessionManager.m [hunk 2]
```diff
@@ -48,7 +48,7 @@ - (void)URLSession:(NSURLSession *)session
         
         NSString *documentsDirectory = [UtilsUrls getOwnCloudFilePath];
         
-        NSString *localCertificatesFolder = [NSString stringWithFormat:@"%@/Certificates/",documentsDirectory];
+        NSString *localCertificatesFolder = [NSString stringWithFormat:@"%@Certificates/",documentsDirectory];
```
Comment (mRoRo) [inline Owncloud iOs Client/Network/URLSession Subclasses/OCURLSessionManager.m]: Same here... getLocalCertificatesPath

File: Owncloud iOs Client/DataBase/Queries/ManageAppSettingsDB.m (hunk 1 above)
Comment (nasli) [inline Owncloud iOs Client/DataBase/Queries/ManageAppSettingsDB.m]: 👍

File: Owncloud iOs Client/Network/URLSession Subclasses/OCURLSessionManager.m (hunk 2 above)
Comment (nasli) [inline Owncloud iOs Client/Network/URLSession Subclasses/OCURLSessionManager.m]: 👍

**Comment by rperezb** [comment]
@mRoRo can you please review this code ?

**Comment by mRoRo** [comment]
@rperezb Sure! I'll have a look

File: Owncloud iOs Client/Utils/UtilsUrls.m [hunk 3]
```diff
@@ -635,4 +635,23 @@ + (NSString *) getFileLocalSystemPathByFileDto:(FileDto *)fileDto andUser:(UserD
     return fullLocalDestiny;
 }
 
+//-----------------------------------
+/// @name getLocalCertificatesPath
+///-----------------------------------
+
+/**
+ * Method used to get only the domain and the protocol (http/https)
```
Comment (mRoRo) [inline Owncloud iOs Client/Utils/UtilsUrls.m]: Change "Method used to get only the domain and the protocol (http/https)" and use the correct method comment.
Same thing for getFileLocalSystemPathByFullPath, getFileLocalSystemPathByFullPath

**Pull request merged**